pkg/user/tx_client_test.go

```diff
@@ -447,3 +451,358 @@ func setupEstimatorService(t *testing.T) *mockEstimatorServer {
 	t.Cleanup(mes.stop)
 	return mes
 }
+
+const (
+	BroadcastTestChainID = "test-broadcast-chain"
+	bufferSize           = 1024 * 1024
+)
+
+type BroadcastTestSuite struct {
+	suite.Suite
+	kr      keyring.Keyring
+	encCfg  encoding.Config
+	signer  *user.Signer
+	account string
+	accAddr sdk.AccAddress
+}
+
+func (s *BroadcastTestSuite) SetupSuite() {
+	s.encCfg = encoding.MakeConfig()
+	s.kr = keyring.NewInMemory(s.encCfg.Codec)
+	s.account = "test_broadcast_account"
+	info, _, err := s.kr.NewMnemonic(s.account, keyring.English, "", "", hd.Secp256k1)
+	s.Require().NoError(err)
+	pubKey, err := info.GetPubKey()
+	s.Require().NoError(err)
+	s.accAddr = sdk.AccAddress(pubKey.Address())
+
+	s.signer, err = user.NewSigner(s.kr, s.encCfg.TxConfig, BroadcastTestChainID, user.NewAccount(s.account, 0, 0))
+	s.Require().NoError(err)
+}
+
+func TestBroadcastTestSuite(t *testing.T) {
+	suite.Run(t, new(BroadcastTestSuite))
+}
+
+// Helper to create a TxClient specifically for broadcast tests, using mock connections.
+func (s *BroadcastTestSuite) setupTestClient(t *testing.T, conns []*grpc.ClientConn) *user.TxClient {
+	t.Helper()
+	require.NotEmpty(t, conns, "Need at least one connection for TxClient")
+
+	primaryConn := conns[0]
+	otherConns := conns[1:]
+
+	txClient, err := user.NewTxClient(
+		s.encCfg.Codec,
+		s.signer,
+		primaryConn,
+		s.encCfg.InterfaceRegistry,
+		user.WithAdditionalCoreEndpoints(otherConns),
+		user.WithDefaultAccount(s.account),
+	)
+	require.NoError(t, err)
+	return txClient
+}
+
+type broadcastTestCase struct {
+	name string
+	// Function to set up mock servers and return their handlers + connections
+	setupMocks        func(t *testing.T) ([]*grpctest.MockTxService, []*grpc.ClientConn, []func())
+	setupContext      func() (context.Context, context.CancelFunc)
+	txOpts            []user.TxOption
+	expectedSuccess   bool
+	expectedHash      string // Only checked if expectedSuccess is true
+	expectedErrSubstr string // Only checked if expectedSuccess is false
+	expectedErrType   error  // Optional: Specific error type check (e.g., context.DeadlineExceeded)
```

Comment:
[question] Why can't this use a single `expectedError error` field and instead needs an expectedErrSubstr and expectedErrType?